File: helpers/dynamodb.py
from boto3 import client
from datetime import datetime

class DynamoDBWrapper:
  def __init__(self):
    self.dynamo_client = client('dynamodb')
    self.ssm_client = client('ssm')
    self.table_name = self.ssm_client.get_parameter(Name='/wikimedia_bot/table_name')['Parameter']['Value']

  def record_post_to_table(self, file_id: int, title: str) -> None:
    """
    Returns:
        None
    """
    self.dynamo_client.put_item(
      TableName=self.table_name,
      Item={
        "file_id": { 'S': str(file_id) },
        "title": { 'S': title },
        "posted_at": { 'S': datetime.now().isoformat() }
      }
    )

  def is_already_posted(self, file_id: int) -> bool:
    response = self.dynamo_client.get_item(
      TableName=self.table_name,
      Key={
        'file_id': {
          'S': str(file_id),
        }
      }
    )

    # If there is a matching item, this field is present.
    # If there is no matching item, this field is absent.
    return 'Item' in response;

File: src/helpers/dynamodb.ts
import {
  DynamoDBClient,
  PutItemCommand,
  GetItemCommand,
} from '@aws-sdk/client-dynamodb';
import { SSMClient, GetParameterCommand } from '@aws-sdk/client-ssm';

export class DynamoDBWrapper {
  private dynamoClient: DynamoDBClient;
  private ssmClient: SSMClient;
  private tableName: string | undefined;

  constructor() {
    this.dynamoClient = new DynamoDBClient({});
    this.ssmClient = new SSMClient({});
  }

  private async getTableName(): Promise<string> {
    if (!this.tableName) {
      const command = new GetParameterCommand({
        Name: '/wikimedia_bot/table_name',
      });
      const response = await this.ssmClient.send(command);
      this.tableName = response.Parameter?.Value;
      if (!this.tableName) throw new Error('table name not found in ssm');
    }
    return this.tableName;
  }

  async recordPostToTable(fileId: number, title: string): Promise<void> {
    const tableName = await this.getTableName();
    const command = new PutItemCommand({
      TableName: tableName,
      Item: {
        file_id: { S: fileId.toString() },
        title: { S: title },
        posted_at: { S: new Date().toISOString() },
      },
    });

    await this.dynamoClient.send(command);
  }

  async isAlreadyPosted(fileId: number): Promise<boolean> {
    const tableName = await this.getTableName();
    const command = new GetItemCommand({
      TableName: tableName,
      Key: {
        file_id: { S: fileId.toString() },
      },
    });

    const response = await this.dynamoClient.send(command);
    return !!response.Item;
  }
}
